1-Draft/RFCnnnn-HTTP POSTing ScriptBlock Logging Provider.md

```diff
@@ -0,0 +1,93 @@
+## Motivation
+From a security professional viewpoint having the ability to monitor PowerShell activity centrally without additional agents and collecting log files is very appealing. Especially the ability to detect and analyse encoded / obfuscated script block executions.
+
+From a DevOp professional viewpoint having the ability to create and view dashboards derived from logs for large scale deployments in real-time would also prove useful to track the status and progress of deployments / builds.
+
+Other benefits include :
+* Reduced OS / Container storage footprint by not installing logging agents etc.
+* Removes complexity of centrally collecting log files e.g. pulling logs from a container instance
+* HTTP POST data is key value pairs making log searching and correlating far more efficient and efficient.
+* No log data is written to disk
+* Removes delay in retrieving logs i.e. The writing to disk followed by an agent polling/pushing the data to the central log aggregation tool
+* As a result reducing the CPU overheads
+
+As an example, [Splunk](https://www.splunk.com) is used internally at the organisation I work at. [Splunk](https://www.splunk.com) offers a [HTTP Collector](http://dev.splunk.com/view/event-collector/SP-CAAAE6M) service which will ingest HTTP POST, raw TCP, raw UDP data making it easily available for searching and reporting centrally near real-time. This service is highly available and will accept 
+
+The inherent benefit of utilising an HTTPS POST method will ensure log data is encrypted during communication and the endpoint URI's identity is validated providing the possibility of posting data over the public internet securely. Great for remote workers!
+
+
+## Specification
+The required data is already made available in the "Microsoft-Windows-PowerShell/Operational" event log.
+### Key Design Considerations
+- The plan would be to create a registry key or a JSON file at system level for configuration to future proof and enable cross platform with PowerShell Core. Potential solution found here - [Dotnet/Corefx file-system based implementation of Microsoft.Win32.Registry](https://github.com/dotnet/corefx/issues/14896).
+- HTTP(S) log posting would require the flexibility of providing either an authentication header (Token based) or provide Basic Authentication mechanism
+- There should be a cmdlet created to convert a user's creds to Basic and encrypted with the machines private key and stored as a secure string. Then at runtime this secure string is decrypted and used in the HTTP headers.
+- Ideally there would be a local cache / queue if the remote endpoint was unavailable for whatever reason. E.g. Remote external workers disconnecting from the VPN.
+
+##### Configuration
+The plan would be to import the configuration data with the following example properties with example values;
+1. StreamLogType
+    * TCP, UDP, HTTP, HTTPS
```

Comment:
How would TCP, UDP, and HTTP data be encrypted?  If we don't have an plan to encrypt, the options should not be included.

Reply:
Perhaps move these to alternatives with notes that we need to figure out how to secure the channels.